subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformValuesInjectionIntegrationTest.groovy

```diff
@@ -212,11 +93,11 @@ project(':b') {
 }
 
 abstract class MakeGreen extends ArtifactTransform {
-    @Inject
-    abstract ArtifactTransformDependencies getDependencies()
+    @PrimaryInputDependencies
+    abstract Iterable<File> getDependencies()
```

Comment:
We can. For me the question is if we should. What would the client do with the file collection apart from iterating the files in it? Or do you think it would be good if there are easier ways to as if the file collection is empty or if it contains a file.

Since we know that the injected file collection is immutable, we at least can be sure it can't be modified by the action.

Reply:
`FileCollection` it is.